light/errors.go

```diff
@@ -20,6 +53,134 @@ func (e ErrOldHeaderExpired) Error() string {
 	return fmt.Sprintf("old header has expired at %v (now: %v)", e.At, e.Now)
 }
 
+type ErrTargetBlockHeightLessThanTrusted struct {
+	Target  int64
+	Trusted int64
+}
+
+func (e ErrTargetBlockHeightLessThanTrusted) Error() string {
+	return fmt.Sprintf("target block has a height lower than the trusted height (%d < %d)", e.Target, e.Trusted)
+}
+
+type ErrHeaderHeightNotMonotonic struct {
+	WantHeight int64
+	GetHeight  int64
```

Comment:
Let's make it more descriptive wrt the case:

```suggestion
	OldHeight int64
	GotHeight int64
```